dali/base/dadfs.cpp

```diff
@@ -8239,20 +8239,22 @@ class CNamedGroupStore: implements INamedGroupStore, public CInterface
 
 };
 
-static CNamedGroupStore *groupStore = NULL;
+static std::atomic<CNamedGroupStore *> groupStore{nullptr};
 static CriticalSection groupsect;
 
 bool CNamedGroupIterator::match()
 {
     if (conn.get()) {
         if (matchgroup.get()) {
-            if (!groupStore)
+            CLeavableCriticalBlock block3(groupsect);
```

Comment:
This critical section will be held over the call to dolookup.  That is likely to cause extra thread contention.

If you are worried about groupStore being destroyed, a more efficient solution is to copy and link it with the critical block, and then use the linked pointer.  However, see note on closedownDFS() below - in which case it can never be freed.